qiskit_experiments/framework/base_experiment.py

```diff
@@ -362,6 +360,11 @@ def circuits(self) -> List[QuantumCircuit]:
         # values for any explicit experiment options that affect circuit
         # generation
 
+    def transpiled_circuits(self):
+        transpile_opts = copy.copy(self.transpile_options.__dict__)
+        transpile_opts["initial_layout"] = list(self.physical_qubits)
+        return transpile(self.circuits(), self.backend, **transpile_opts)
```

Comment:
post transpile hook should be called here

Reply:
Perhaps. But this is not related to this PR. The code as it is now is equivalent to what was before - we first transpile all the circuits, then we call the post-process method of the root experiment.

Reply:
By the way this is related: https://github.com/Qiskit/qiskit-experiments/issues/666

Reply:
> then we call the post-process method of the root experiment.
I don't think so. post transpile is defined per each sub experiment. this will be much more important with above PR.

Reply:
The code that's currently in the main branch calls the post-process method of the root experiment.
https://github.com/Qiskit/qiskit-experiments/blob/fc824d671426d8943e735551b15790b9f5f26c28/qiskit_experiments/framework/base_experiment.py#L273

If this is not fine then it should be fixed in a separate PR (#677 actually). The scope of the batch PR (#667) is to transpile by sub-experiment transpile options. Everything else around this is being left equivalent to the original code, and should be modified in separate PRs.

Reply:
I think leaving this to be removed in 677 is fine